### PR TITLE
Hastebin Conversion

### LINKED ISSUE #384: Convert log files into a pastebin post on the discord.
### Feature request

**Checklist**
- [x] I have read and agree to the README (found on https://github.com/Minehut/Meta)
- [x] This is not a bug report, support request or plugin request
- [x] I have searched the Issue Tracker to make sure this is not a duplicate issue
- [x] This issue does not contain any sensitive information 

**Platform**
DISCORD

**Describe the feature**

I often see people send in their logs as a `message.txt` file when using #support, and this is just a hassle for everybody especially when the user helping is on mobile as they have to download the file, and figure out a way to view it. 

My suggestion is that the Minehut Bot detects when a log file is sent and automatically creates a pastebin post and sends it to the channel. Pastebin itself has an easy to use and well documented API so it shouldn't be too hard to make.

This would be implemented in the #support channel, however could work in #skript & #code.

The GeyserMC discord has this feature and I've seen first hand just how helpful it can be. 
Screenshot_20201115-193938_Discord

**How the feature is useful**

#### This is super handy for anybody trying to help another player by viewing their logs easily and effortlessly.
---
**Use 👍 and 👎 reactions to indicate you agree or disagree with this suggestion! Feel free to discuss the idea in the comments.**

### PR DESCRIPTION
**Pull request by Sniped**
Converts message attachments with a configured whitelisted extension to a hastebin link and sends it when such a message attachment is sent in a configured channel. It also has a cooldown of 10 seconds to ensure people can't spam attachments.

Closes https://github.com/Minehut/Meta/issues/384

### CODE REVIEW

File: src/guild/config/guildConfigs.ts [hunk 1]
```diff
@@ -73,6 +73,10 @@ guildConfigs.set('239599059415859200', {
 				{ roleId: '493253127366115360', emoji: 'HypeBadge' },
 			],
 		},
+		hastebinConversion: {
+			channels: ['412394499919052810', '660337933743816724', '400170127737356299'],
+			whitelistedExtensions: ['txt']
```
Comment (theminecoder) [inline src/guild/config/guildConfigs.ts]:
```suggestion
			whitelistedExtensions: ['txt', 'log', 'json', 'yml', 'yaml', 'properties', 'sk']
```

**Review by theminecoder — COMMENTED** [review]
Just adding a few extensions before this goes through

File: src/listener/hastebinConversion.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+			.hastebinConversion;
+		if (
+			!hastebinConversionConfig ||
+			!hastebinConversionConfig.channels.includes(msg.channel.id)
+		)
+			return;
+
+		const messageAttachment = msg.attachments.find(attachment =>
+			hastebinConversionConfig.whitelistedExtensions.some(ext =>
+				attachment.name?.endsWith(ext)
```
Comment (theminecoder) [inline src/listener/hastebinConversion.ts]:
```suggestion
				attachment.name?.endsWith(`.${ext}`)
```

Making sure this is actually an extension

**Review by theminecoder — APPROVED** [review]

**Comment by AdxmGG** [comment]
Can I suggest for this to look a bit nicer? I think this is already 100x nicer than just showing the raw link. 
image

**Comment by Sniped** [comment]
> Can I suggest for this to look a bit nicer? I think this is already 100x nicer than just showing the raw link.
> image

This pull request doesn’t only target logs anymore. It targets skript files, yaml configurations, and properties files as well.

**Pull request merged**